x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java

```diff
@@ -154,6 +154,7 @@ public class ClusterPrivilegeResolver {
     private static final Set<String> READ_SLM_PATTERN = Set.of(GetSnapshotLifecycleAction.NAME, GetStatusAction.NAME);
 
     private static final Set<String> MANAGE_SEARCH_APPLICATION_PATTERN = Set.of("cluster:admin/xpack/application/search_application/*");
+    private static final Set<String> MANAGE_SEARCH_SYNONYMS_PATTERN = Set.of("cluster:admin/synonyms/*", "cluster:admin/synonyms_sets/*");
```

Comment:
We want to make sure that we have `search` in the pattern for future "manage search" privileges.

Also, does this synonyms need two different permission patterns? I would think that if you have the permission to manage synonym sets, you have the permission to manage synonym rules, and vice versa. Would it make sense to simplify this to just something like `cluster:admin/search/synonyms`?

Reply:
Do we need to add `admin/synonym_rules/*`?

Reply:
> We want to make sure that we have search in the pattern for future "manage search" privileges.

Is that needed? I think we can use something like [CROSS_CLUSTER_REPLICATION](https://github.com/elastic/elasticsearch/blob/ac9822be7247d33ad29ced39ae2eb5e4d3fcd189/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java#L163) where we use a set of privileges, and we don't need explicit naming prefix...

I think it's better to omit the `search` prefix as that it's not a pattern I see in other constants, and we don't need to have it in order to create a `manage_search` privilege. Not a strong opinion though, happy to change it.

> Also, does this synonyms need two different permission patterns? I would think that if you have the permission to manage synonym sets, you have the permission to manage synonym rules, and vice versa. Would it make sense to simplify this to just something like cluster:admin/search/synonyms?

We have three different action types, regarding synonyms, synonyms sets and synonym rules. As managing synonyms grant you access to both, that's the meaning of using the set of actions here.

> Do we need to add admin/synonym_rules/*?
🤦  yes, thanks!

Reply:
@kderusso , @mayya-sharipova , anything else that you want addressed here? Hopefully my responses above and the changes to add `admin/synonym_rules/*` did address your concerns. Please LMK!